aws/transport/http/response_error_middleware.go

```diff
@@ -0,0 +1,65 @@
+package http
+
+import (
+	"context"
+	"errors"
+
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+// AddResponseErrorWrapper adds response error wrapper middleware
+func AddResponseErrorWrapper(stack *middleware.Stack) {
+	// add error wrapper middleware before operation deserializers so that it can wrap the error response
+	// returned by operation deserializers
+	stack.Deserialize.Insert(&errorWrapperMiddleware{}, "OperationDeserializer", middleware.Before)
+}
+
+// RequestID is the identifier for Request Id set in response metadata
+const RequestID = "RequestID"
```

Comment:
probably want to use getter/setter helpers for the metadata that use struct keys instead of strings. This way we can ensure that there are no collisions with names.  similar to how the SDK is using context to get and store values.